### PR TITLE
Add password authentication to KeepalivedGroup

### PR DESCRIPTION
**Pull request by tommasopozzetti**
As requested in #48, this PR introduces the ability to set up password authentication for VRRP instances. The configuration is global at the KeepalivedGroup level, therefore all VRRP instances managed by pods in the given KeepalivedGroup will share the same password.

The password can be specified via a secret that can be referenced in the KeepalivedGroup CR as follows:
```yaml
apiVersion: redhatcop.redhat.io/v1alpha1
kind: KeepalivedGroup
metadata:
  name: keepalivedgroup-router
spec:
  image: registry.redhat.io/openshift4/ose-keepalived-ipfailover
  interface: ens3
  nodeSelector:
    node-role.kubernetes.io/loadbalancer: ""
  passwordAuth:
    secretRef:
      name: my-secret
    secretKey: vrrpPass
---

apiVersion: v1
kind: Secret
metadata:
  name: my-secret
stringData:
  vrrpPass: 1234
```
The `secretKey` value is not mandatory and defaults to `password` if not specified.

### CODE REVIEW

File: controllers/keepalivedgroup_controller.go [hunk 1]
```diff
@@ -145,6 +146,23 @@ func (r *KeepalivedGroupReconciler) Reconcile(context context.Context, req ctrl.
 		return reconcile.Result{}, nil
 	}
 
+	// Check if VRRP authentication is needed and if so extract credentials
+	authPass := ""
+	if instance.Spec.PasswordAuth.SecretRef.Name != "" {
+		secret := &corev1.Secret{}
+		err := r.GetClient().Get(context, types.NamespacedName{Namespace: instance.GetNamespace(), Name: instance.Spec.PasswordAuth.SecretRef.Name}, secret)
+		if err != nil {
+			// Requeue and log error
```
Comment (raffaelespazzoli) [inline controllers/keepalivedgroup_controller.go:155]: please follow the pattern for the other error management code blocks, ie:
1. log the error
2.  return with the manage error function.

File: controllers/keepalivedgroup_controller.go [hunk 2]
```diff
@@ -145,6 +146,23 @@ func (r *KeepalivedGroupReconciler) Reconcile(context context.Context, req ctrl.
...
+			return reconcile.Result{}, err
+		}
+		pass, ok := secret.Data[instance.Spec.PasswordAuth.SecretKey]
+		if !ok {
+			// Requeue and log error
```
Comment (raffaelespazzoli) [inline controllers/keepalivedgroup_controller.go:160]: same as above.

**Review by raffaelespazzoli — CHANGES_REQUESTED** [review]

**Comment by tommasopozzetti** [comment]
@raffaelespazzoli thanks for reviewing this.
I have changed the error handling to conform to the suggested pattern.

**Comment by David-Igou** [comment]
LGTM

**Pull request merged**